src/schema.rs

```diff
@@ -7,13 +7,13 @@ use std::path::Path;
 /// Parse a given file and convert it to Schema object
 pub trait FromFile {
     /// Parses a given file path into a a valid Schema object
-    fn parse_file(path: &Path) -> Result<Schema, DegaussError>;
+    fn parse_file<P: AsRef<Path>>(path: P) -> Result<Schema, DegaussError>;
 }
 
 /// Implements the FromFile trait for reading Schema from a given file
 impl FromFile for Schema {
     /// Parses a given file into a a valid Schema object
-    fn parse_file(path: &Path) -> Result<Schema, DegaussError> {
+    fn parse_file<P: AsRef<Path>>(path: P) -> Result<Schema, DegaussError> {
```

Comment:
Uh I thought this is coming from avro-rs. :D Hahaha nice. Got rid of `PathBuf::from_str`